_posts/2020-09-21-dsc-release.md

```diff
@@ -0,0 +1,199 @@
+
+We picked several of the most-downloaded modules with DSC Resources released in the last year to start with, but in the coming weeks you should find _everything_ available on the PowerShell Gallery up on the Forge!
+
+But today, in addition to sharing this news, I want to go into a bit more detail, both with what's going on under the hood and with how you can use these auto-generated modules.
+
+### Installing the Modules
+
+If you just want to get your hands on the PowerShell module to read the help and experiment on your own, you can install it like any other PowerShell module:
+
+```powershell
+Install-Module -Name Puppet.Dsc
+Import-Module -Name Puppet.Dsc
+Get-Command -Module Puppet.Dsc
+Get-Help -Name New-PuppetDscModule -ShowWindow
+```
+
+Some of the functions from the PowerShell module rely on the [Puppet Development Kit (PDK)][pdk-docs] being installed;
+we suggest using [Chocolatey][choco-site] to do so, but you can also [manually download and install][pdk-download] if you prefer.
+
+```bash
+choco install pdk
+```
+
+If you want to do the same sort of thing with one of the Puppetized modules from the forge, you can install them and treat them just like any other Puppet module:
+
+```bash
+# For example, installing the Puppetized ComputerManagementDsc module manually:
+puppet module install dsc-computermanagementdsc --Version 8.4.0-0-0
+# Or via r10k / Code Manager / Bolt:
+mod 'dsc-computermanagementdsc', '8.4.0-0-0'
+```
+
+## Puppetizing a PowerShell module
+
+So, you want to turn a PowerShell module with DSC Resources into a Puppet module?
+You can do this with a single command:
+
+```powershell
+# If you run this command without the PassThru flaf it returns nothing to the output stream.
+# If you'd like a deeper look at what it's doing, try running it with the Verbose flag.
+New-PuppetDscModule -PowerShellModuleName ComputerManagementDsc -PassThru
+```
+
+Unfortunately, the output from running it isn't all that exciting:
+
+![Output from a PowerShell console showing the running of the command above as well as the return output, which is the folder object representing the new Puppet module on disk][img-new-puppetdscmodule]
+
+Behind the scenes, the command:
+
+1. Uses the [PDK][pdk-docs] to create a new Puppet module
+1. Searches the [PowerShell Gallery][psgallery] for the ComputerManagementDsc module and vendor it into the new Puppet module
+1. Converts each of the DSC Resources from the vendored PowerShell module into a Puppet Resource API type and provider
+1. Updates the Puppet module's metadata and documentation based on the PowerShell module.
+
+> At this time, for best results we strongly recommend you run the command with administrator privileges in a Windows PowerShell 5.1 session.
+> While it _does_ work without administrator privileges, the function cannot fully map nested CIM instances without administrator privileges.
```

Comment:
Is there a technical reason for the inability to map nested CIM that should be explained, or is that something that a powershell user will just know?

Reply:
There _is_, but it's covered in the linked help file on the next line; tldr the method for crawling them requires running the DSC resource once to load it in memory which needs admin privs.

Reply:
👍